client/src/utils/responsiveness.ts

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from 'react'
+
+/**
+ * Given a CSS media query, returns whether the media query is matched, e.g.
+ *
+ * const isMobileWidth = useMediaQuery("(max-width: 48em)")
+ *
+ * Listens for updates, e.g. in response to window resizing.
+ */
+export function useMediaQuery(query: string): boolean {
+  const [isMatched, setIsMatched] = useState(false)
+
+  useEffect(() => {
+    const matchMedia = window.matchMedia(query)
+    setIsMatched(matchMedia.matches)
+    const handleChange = () => {
+      setIsMatched(matchMedia.matches)
+    }
+    matchMedia.addEventListener('change', handleChange)
+    return () => {
+      matchMedia.removeEventListener('change', handleChange)
+    }
+  }, [query])
+
+  return isMatched
+}
+
+export const BREAKPOINT_MOBILE = '40em'
+export const BREAKPOINT_DESKTOP = '64em'
```

Comment:
Oh the art of picking breakpoints. I consulted a few articles (there are hundreds) and ultimately took inspiration from Clever's design system!

[`@breakpointS: 40em;`](https://github.com/Clever/components/blob/454320efdc04e432e1aaa0f38656ba702fa77084/src/less/layout.less#L6-L10)
[`@breakpointL: 64em;`](https://github.com/Clever/components/blob/454320efdc04e432e1aaa0f38656ba702fa77084/src/less/layout.less#L26-L30)